### DISCUSSION TITLE
NEW VERSION TEST

### DISCUSSION THREAD

**Discussion by nagadit** [nagadit/DeepFaceLab_Linux#61] [General]
Here we discuss the general results of script tests on different systems

**Comment by CeeBeeEh** [comment]
Can I suggest a docker based implementation? That would solve a great many dependency issues. I'm using Ubuntu 18.04 on my desktop and I continually run into issues with qt5. On 18.04 apt provides version 5.10 but pip provides 5.15 which creates a conflict as some applications (haven't hunted them down) look for the apt version.

I've created a [fork of DeepFaceLive ](https://github.com/CeeBeeEh/DeepFaceLive) to address some of the issues with the Linux version (use my dev branch). Currently the changes I made allow the docker version to work extremely well with little to no configuration. 

I realize that using docker would limit users of Unixes such as FreeBSD.

I have a 1080ti and an RTX Titan that I can test on.

**Reply by nagadit** [reply]
Docker is cool, but on some servers where sudo is not available and docker is not available, using conda will solve problems.

My repository implies a completely manual setup and use.  I personally changed the DFLab code several times.

**Reply by Pheidologeton** [reply]
Flatpak is a good distribution option. It has the same performance as native packages or docker, but is easier to support, install and upgrade.

**Comment by Pheidologeton** [comment]
Могу протестировать на arch linux + gtx 1080ti. Так же было бы неплохо прикрутить простенький gui (ну или tui) и добавить в AUR

**Comment by zhangweilst** [comment]
1. Seems like the script `4.1) data_src view aligned result.` is missing on the unix branch, which is a handy tool after sorting the faces.
2. I maybe can help to test scripts on Ubuntu 20.04 with RTX 3060, just let me know if necessary.

**Comment by gteachey** [comment]
Hello @nagadit!

Not sure if this is still being worked on, here's my setup:
OS: Pop OS! 22.04 LTS
GPU: NVIDIA RTX 3060
RAM: 32 GB


I'm running on Python3.9, CUDA toolkit 11.8, CUDNN 8.6
Full list of my conda environment:
[deepfacelab.txt](https://github.com/nagadit/DeepFaceLab_Linux/files/12137263/deepfacelab.txt)

I ran into a conflict with cv2 and PyQt5. It was messing up running any scripts for XSegEditor or showing the training window(6_train*.sh) scripts.
 I forgot to copy mine but it's the same as this one:
```
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "$HOME/Documents/gui_nn/venv/lib/python3.8/site-packages/cv2/qt/plugins" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: xcb, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, webgl.

```
To fix it I specified to use the PyQt5 path in the DeepFaceLab_Linux/DeepFaceLab/XSegEditor/XSegEditor.py file by adding these lines to it:
```
import cv2
import numpy as np
import numpy.linalg as npla
**import PyQt5**
from PyQt5.QtCore import *

```

then down by around line 1458, added code to find the PyQt5 plugins folder and set it to the QT_QPA_PLATFORM_PLUGIN_PATH environment variable
`os.environ["QT_QPA_PLATFORM_PLUGIN_PATH"] = os.fspath(
    Path(PyQt5.__file__).resolve().parent / "Qt5" / "plugins"
    )`
